### PR TITLE
Feature/cf 16

### PR DESCRIPTION
**Pull request by dinocom33**
This PR represents the implementations of all DB models as follows:

- game(represents all games like Yogioh, Star Wars etc.)
- yugioh(represents Yugioh card details)
- yugiohcardinset(represent connection between yugioh - yugiohcardrarity - yugiohcardset)
- yugiohcardrarity(represent card rarity)
- yugiohcardset(represent card set)
- YugiohCardYugiohCardInSet - This model was created to hold relations between yugioh card and yugioh card in set, so there will be only one card object created instead of creating a separate card for each card set.

Also Swagger UI has benn installed to make API testing easier and maintain API documentation.

### CODE REVIEW

File: backend/yugiohcardrarity/models.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        max_length=100,
+        unique=True,
+    )
+
+    class Meta:
+        verbose_name_plural = 'Yugioh Card Rarities'
+        verbose_name = 'Yugioh Card Rarity'
+
+    def __str__(self):
+        return {self.rarity}
```
Comment (julkascript) [inline backend/yugiohcardrarity/models.py:15]: `__str__` method requires a string to be returned. In this case we are returning a set, which raises a TypeError upon creation. 

- [ ] Fix return value to be string

File: backend/yugiohcardrarity/models.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from django.db import models
+
+
+class YugiohCardRarity(models.Model):
+    rarity = models.CharField(
+        max_length=100,
+        unique=True,
+    )
+
```
Comment (julkascript) [inline backend/yugiohcardrarity/models.py:9]: - [ ] Add rarity `code` field

Exemplary set and code:
`Phantom nightmare` - `PHNI`

File: backend/yugiohcardrarity/models.py (hunk 1 above)
Comment (dinocom33) [reply backend/yugiohcardrarity/models.py]: fixed

File: backend/yugiohcardset/models.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from django.db import models
+
+from yugiohcardinset.models import YugiohCardInSet
+
+
+class YugiohCardSet(models.Model):
+    card_set_name = models.CharField(
+        max_length=100,
+        unique=True,
+    )
```
Comment (julkascript) [inline backend/yugiohcardset/models.py:10]: - [ ] Add `code` field

The code is unique. Exemplary code and set would be: `Phantom Nightmare` - `PHNI`

File: backend/yugioh/models.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    card = models.ForeignKey(
+        'card.Card',
+        on_delete=models.CASCADE,
+    )
+
+    def __str__(self):
+        return self.card
+
+
+class YugiohCardYugiohCardInSet(models.Model):
```
Comment (julkascript) [inline backend/yugioh/models.py:70]: - [ ] Rename to `YugiohCardInSet`

Comment (dinocom33) [reply backend/yugioh/models.py:70]: We already have YugiohCardInSet model

File: backend/yugiohcardset/models.py (hunk 3 above)
Comment (dinocom33) [reply backend/yugiohcardset/models.py:10]: done

File: backend/yugioh/models.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        null=True,
+    )
+
+    card = models.ForeignKey(
+        'card.Card',
+        on_delete=models.CASCADE,
+    )
+
+    def __str__(self):
+        return self.card
```
Comment (julkascript) [inline backend/yugioh/models.py:67]: This returns the `Card` object, which is **not a string**, and therefore throws a `TypeError`. The `__str__` method requires a string to be returned.

- [ ] Make the `__str__` method return the card name (str)

Comment (dinocom33) [reply backend/yugioh/models.py:67]: fixed

File: backend/yugioh/models.py (hunk 4 above)
Comment (dinocom33) [reply backend/yugioh/models.py:70]: YugiohCard and YugiohCardInSet have been added to admin panel.

**Comment by julkascript** [comment]
For now it is okay if all `Yugioh...` models are in the yugioh app. I don't think our project will get that so big that it requires that much granularity. 

- [ ] Move `YugiohCard, YugiohCardRarity, YugiohCardSet, YugiohCardInSet` models to the yugioh app. Remove the respective apps and usages.

**Comment by julkascript** [comment]
Always aim to have 1 migration file per PR. Otherwise, we would stack a lot of needless migrations.

- [ ] Squash or remove all migrations and leave just 1 migration per app.

File: backend/yugioh/models.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        return f'{self.set} - {self.rarity}'
+
+
+class YugiohCardSet(models.Model):
+    card_set_name = models.CharField(
+        max_length=100,
+        unique=True,
+    )
+
+    rarity_code = models.CharField(
```
Comment (julkascript) [inline backend/yugioh/models.py:93]: - [ ] Rename to `set_code`

Comment (dinocom33) [reply backend/yugioh/models.py]: The change has been made

File: backend/yugioh/admin.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+@admin.register(YugiohCard)
+class YugiohCardAdmin(admin.ModelAdmin):
+    list_display = ['type', 'frame_type', 'description', 'attack', 'defense', 'level', 'race', 'attribute', 'archetype']
+    list_filter = ['type', 'frame_type', 'description', 'attack', 'defense', 'level', 'race', 'attribute', 'archetype']
+    search_fields = ['type', 'frame_type', 'description', 'attack', 'defense', 'level', 'race', 'attribute', 'archetype']
+
+
+@admin.register(YugiohCardInSet)
+class YugiohCardInSetAdmin(admin.ModelAdmin):
```
Comment (julkascript) [inline backend/yugioh/admin.py:14]: - [ ] Add the card name as the first column

File: backend/yugioh/models.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        on_delete=models.CASCADE,
+    )
+
+    yugioh_card = models.ForeignKey(
+        YugiohCard,
+        on_delete=models.CASCADE,
+    )
+
+    def __str__(self):
+        return f'{self.set} - {self.rarity}'
```
Comment (julkascript) [inline backend/yugioh/models.py:84]: - [ ] Represent as `card name` - `set` - `rarity`

File: backend/yugioh/admin.py (hunk 7 above)
Comment (dinocom33) [reply backend/yugioh/admin.py:14]: The name was added

File: backend/yugioh/models.py (hunk 8 above)
Comment (dinocom33) [reply backend/yugioh/models.py:84]: The new represent was set.

File: backend/yugioh/models.py [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+        blank=True,
+        null=True,
+    )
+
+    image = models.URLField(
+        blank=True,
+        null=True,
+    )
+
+    card = models.ForeignKey(
```
Comment (julkascript) [inline backend/yugioh/models.py:60]: The `YugiohCard` should have a 1:1 relationship with the `Card` model, as per the diagram. I advise you to make the `YugiohCard` model inherit the card model. After that, fix the `__str__` representation.

Comment (dinocom33) [reply backend/yugioh/models.py]: All necessary changes have been made.

**Review by julkascript — APPROVED** [review]

**Pull request merged**